File: employee-system-api/src/main/java/com/rajkumar/springboot/service/EmployeeService.java
/**
 * 
 */
package com.rajkumar.springboot.service;

import java.util.List;

import com.rajkumar.springboot.model.Employee;

/**
 * @author Rajkumar. S
 *
 */
public interface EmployeeService {
    
    Employee createEmployee(Employee employee);
    
    List<Employee> getAllEmployees();
    
    boolean deleteEmployee(String id);
    
    Employee getEmployeeById(String id);
    
    Employee updateEmployee(String id, Employee emp);
    
}

File: employee-management-system-ui/src/services/EmployeeService.js
import axios from 'axios';

const EMPLOYEE_API_BASE_URL = 'http://localhost:8080/api/v1/employees';

class EmployeeService {
  saveEmployee(employee) {
    return axios.post(EMPLOYEE_API_BASE_URL, employee);
  }

  getAllEmployees() {
    return axios.get(EMPLOYEE_API_BASE_URL);
  }

  deleteEmployee(id) {
    return axios.delete(`${EMPLOYEE_API_BASE_URL}/${id}`);
  }

  getEmployeeById(id) {
    return axios.get(`${EMPLOYEE_API_BASE_URL}/${id}`);
  }

  updateEmployee(id, emp) {
    return axios.put(`${EMPLOYEE_API_BASE_URL}/${id}`, emp);
  }
}

export default new EmployeeService();